src/Core/System/SalesChannel/Context/SalesChannelContextService.php

```diff
@@ -94,6 +94,13 @@ public function get(SalesChannelContextServiceParameters $parameters): SalesChan
             }
 
             $context = $this->factory->create($token, $parameters->getSalesChannelId(), $session);
+
+            if ($originalStates = $parameters->getOriginalContext()?->getStates()) {
```

Comment:
I think this could lead to unexpected behavior, maybe we can check explicitly the preview search state from original and add to the new context